client/components/stats-date-control/stats-date-control-picker-shortcuts.tsx

```diff
@@ -1,40 +1,40 @@
 import { Button } from '@wordpress/components';
 import { Icon, check } from '@wordpress/icons';
+import classNames from 'classnames';
 import React from 'react';
-import { DateControlPickerShortcut, DateControlPickerShortcutsProps } from './types';
+import { DateControlPickerShortcutsProps } from './types';
 
 const DateControlPickerShortcuts = ( {
 	shortcutList,
 	currentShortcut,
 	onClick,
 }: DateControlPickerShortcutsProps ) => {
-	// Simple selection test.
-	function isSelectedShortcut( shortcut: DateControlPickerShortcut ) {
-		return shortcut.id === currentShortcut;
-	}
-	// Apply selection state via CSS.
-	function classNameForShortcut( shortcut: DateControlPickerShortcut ): string {
-		const defaultClassName = 'date-control-picker-shortcuts__shortcut';
-		const selectedClassName = defaultClassName + ' is-selected';
-		return isSelectedShortcut( shortcut ) ? selectedClassName : defaultClassName;
-	}
-
 	return (
 		<div className="date-control-picker-shortcuts">
 			<ul className="date-control-picker-shortcuts__list">
-				{ shortcutList.map( ( shortcut, idx ) => (
-					<li className={ classNameForShortcut( shortcut ) } key={ shortcut.id || idx }>
-						<Button
+				{ shortcutList.map( ( shortcut, idx ) => {
+					const isSelected = shortcut.id === currentShortcut;
+
+					return (
+						<li
+							className={ classNames( 'date-control-picker-shortcuts__shortcut', {
```

Comment:
Hi @a8ck3n,

Just FYI I think it's simpler and easier to understand if written this way. `classNames` is so well known for Reactjs developers, and is kinda a standard.

I always appreciate readability over simplicity. If it's simple and readable then it's better than verbosity and readable. Verbosity means more code to write and read - IMHO in most cases, it doesn't have as good readability. 😄

Reply:
Agreed on the use of `classNames` here, although I don't feel strongly about it.

I disagree when it comes to inlining logic in the JSX though. As someone newer to JSX, I definitely find it harder to parse when mixing logic and presentation like this. I'd say the same for anonymous functions in general. When they move beyond a single line or two, they feel like an abuse of the pattern. I'd like to be able to update logic without accidentally altering layout which is easier to reason about when having clearer separation.

I also find it harder to parse DIFFs when we inline our logic. Partially this is due to the auto-formatting we apply but it means that as things indent, they are more likely to be reformatted and pollute the resulting DIFF.